lib/mongo/cluster/reapers/cursor_reaper.rb

```diff
@@ -56,12 +62,14 @@ def initialize(cluster)
       #
       # @api private
       def schedule_kill_cursor(kill_spec, server)
-        @mutex.synchronize do
-          if @active_cursor_ids.include?(kill_spec.cursor_id)
-            @to_kill[server.address.seed] ||= Set.new
-            @to_kill[server.address.seed] << kill_spec
-          end
-        end
+        msg = [
+          server.address.seed,
+          kill_spec.cursor_id,
+          kill_spec.coll_name,
+          kill_spec.db_name,
+          kill_spec.service_id
+        ].join(MSG_SEPARATOR)
+        @pipe_write.puts(msg)
```

Comment:
This call can block if the pipe is full. My understanding is there needs to be  a thread that is constantly reading the pipe to ensure the writing side doesn't block indefinitely.

Reply:
Have you considered https://ruby-doc.org/core-2.5.0/Queue.html ?

Reply:
I have, but `Queue` is thread safe, and according to its docs _"The [Queue](https://ruby-doc.org/core-2.5.0/Queue.html) class implements all the required locking semantics."_ So, we may encounter the same problem with queue. I'll do some testing, though. If it works, I would prefer queue over pipe here.